qulice-checkstyle/src/main/java/com/qulice/checkstyle/QuliceIndentationCheck.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2011-2015, Qulice.com
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met: 1) Redistributions of source code must retain the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer. 2) Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the Qulice.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.qulice.checkstyle;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck;
+import java.io.File;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Make sure each line indentation is either:
+ * <ul>
+ * <li>the same as previous one or less
+ * <li>bigger than previous by exactly 4
+ * </ul>
+ * All other cases must cause a failure.
+ *
+ * @author Hamdi Douss (douss.hamdi@gmail.com)
+ * @version $Id$
+ */
+public final class QuliceIndentationCheck extends AbstractFileSetCheck {
+    /**
+     * {@inheritDoc}
+     * @checkstyle MagicNumberCheck (13 lines)
+     */
+    @Override
+    public void processFiltered(final File file, final List<String> lines) {
+        int previous = 0;
+        for (int pos = 0; pos < lines.size(); pos += 1) {
+            final String line = lines.get(pos);
+            final int current = QuliceIndentationCheck.indentation(line);
+            if (QuliceIndentationCheck.inCommentBlock(line)
+                || line.length() == 0) {
+                continue;
+            }
+            if (current > previous
+                && current != previous + 4) {
```

Comment:
@HDouss put `4` in a constant and then you can reuse it in the string below, and remove the `MagicNumberCheck` exclude